__to_graphql_input throws assertion error when mutation argument contains a list of objects

The schema we use has a mutation that requires an Input object:
```
mutation(input: {
  id: "test_id"
  inputSlots: [
    {
      field1: ""
      field2: ""
    },
    {
      field1: ""
      field2: ""
    }
  ]

```

When performing operation.__to_graphql__() I got an error: "AssertionError: InputSlot(field1="", field2="") is not a JSON Object". With version 13.0 we don't have this issue.
The error is thrown in types\__init__.py line 1747 in class ContainerType
![image](https://user-images.githubusercontent.com/25790249/131341414-13dd8c67-a98c-4471-bbb2-0b4b8ea21498.png)

The reason why this happend is becuase inputSlots is of type [InputSlot]! and the library converted this before to this type. Thus the dict values in the array where already converted to their corresponding type and are not of type dict anymore.

--- a/sgqlc/types/__init__.py
+++ b/sgqlc/types/__init__.py
@@ -1000,7 +1000,7 @@ def _create_list_of_wrapper(name, t):
         if json_data is None:
             return None
 
-        if isinstance(json_data, (t, Variable)):  # pragma: no cover
+        if isinstance(json_data, (t, Variable)):
             return json_data
 
         return [realize_type(v, selection_list) for v in json_data]
@@ -1157,7 +1157,7 @@ def list_of(t):
       ``[1, 2]``;
 
     - ``non_null_list_of_non_null_int`` means it must be a list, not
-      ``None`` **and** the list elements must not be ``Non``, ie:
+      ``None`` **and** the list elements must not be ``None``, ie:
       ``[1, 2]``.
 
     >>> class TypeWithListFields(Type):
@@ -1282,6 +1282,23 @@ def list_of(t):
     >>> print(json.dumps(list_of(int).__to_json_value__(None)))
     null
 
+    Lists can be of complex types, for instance :class:`Input`:
+
+    >>> class SomeInput(Input):
+    ...     a = int
+    >>> SomeInputList = list_of(SomeInput)
+    >>> SomeInputList([{'a': 123}])
+    [SomeInput(a=123)]
+
+    Variables may be given as constructor parameters:
+
+    >>> SomeInputList(Variable('lst'))
+    $lst
+
+    Or already realized lists:
+
+    >>> SomeInputList(SomeInputList([{'a': 123}]))
+    [SomeInput(a=123)]
     '''
     if isinstance(t, str):
         return Lazy(t, '[' + t + ']', list_of)

